.github/workflows/benchmarks.yml

```diff
@@ -0,0 +1,50 @@
+name: Run Benchmarks
+
+# on:
+#   schedule:
+#     - cron: '0 0 * * 0'  # Runs weekly on Sunday at midnight
+on:
+  pull_request:
+    
+jobs:
+  benchmark:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: fetch submodules
+        run: git submodule update --init --recursive
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          toolchain: nightly
+          target: wasm32-wasi
+      - name: Setup build env
+        run: |
+          os=$(echo "$RUNNER_OS" | tr '[:upper:]' '[:lower:]')
+          ./scripts/setup-$os.sh
+        shell: bash
+      - name: Run Benchmarks
+        run: cargo +nightly bench -p containerd-shim-benchmarks -- --output-format bencher  | tee output.txt
+      # Download previous benchmark result from cache (if exists)
+      - name: Cache benchmark data
+        uses: actions/cache@v4
+        with:
+          path: ./cache
+          key: ${{ runner.os }}-benchmark
+      - name: Store benchmark result
+        uses: benchmark-action/github-action-benchmark@v1.20.3
+        with:
+          name: Criterion.rs Benchmark
+          tool: 'cargo'
+          output-file-path: output.txt
+          # GitHub API token to make a commit comment
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # Show alert with commit comment on detecting possible performance regression
+          alert-threshold: '200%'
```

Comment:
This seems pretty big, how did you select this number?  Can we trigger it in CI to make sure this sets a comment?

Reply:
That does seem pretty big. I set the number randomly without much thoughts. 

I would prematurely set it to 150% for now since the numbers flutuate +- 20% in my experience of running runwasi benchmarking locally. 

> Can we trigger it in CI to make sure this sets a comment?
Sure, I can try setting it to a lower number and test it works.

Reply:
Example: https://github.com/containerd/runwasi/actions/runs/9469368717/job/26087869930#step:8:29

Reply:
20% seems like a big fluctuation but maybe the actual times are pretty small. Getting this into a graph format long term would be helpful to see these types of trends.

>https://github.com/containerd/runwasi/actions/runs/9469368717/job/26087869930#step:8:29 

cool!  does it also post to the PR? Maybe another reason to make sure we split this out into individual runs.

Reply:
>this tool seems to be able to support graphs https://github.com/benchmark-action/github-action-benchmark?tab=readme-ov-file#charts-on-github-pages

Yeah, we don't have a GitHub Pages on this repo yet so I have not tried this feature. That could be a follow-up.

Reply:
#620